pkgs/servers/pingvin-share/default.nix

```diff
@@ -0,0 +1,35 @@
+{ callPackage
+, fetchFromGitHub
+, nixosTests
+}:
+
+let
+  version = "0.22.0";
+  src = fetchFromGitHub {
+    owner = "stonith404";
+    repo = "pingvin-share";
+    rev = "v${version}";
+    hash = "sha256-vQUn2yRelj4j9q5Rkqz305b7Vedv3CSpOh5HMwhWJxs=";
```

Comment:
It would be easier and the update bot would probably just work, if version and src would be part of a package, eg the backend and this file would contain a real package or packageset

Reply:
I didn't understand this comment, do you have an example ?

Reply:
If we move this into the backend package instead of having it in the let in here, then I think the update bot would just work.

Reply:
@SuperSandro2000 is this a blocker? I'd like to help move this forward because I think it's a well made package.

Would something like this be ok? 

```
diff --git a/pkgs/servers/pingvin-share/backend.nix b/pkgs/servers/pingvin-share/backend.nix
index ef8efd7354bd..c6ffaad1c9ca 100644
--- a/pkgs/servers/pingvin-share/backend.nix
+++ b/pkgs/servers/pingvin-share/backend.nix
@@ -1,21 +1,27 @@
 { lib
+, fetchFromGitHub
 , buildNpmPackage
 , vips
 , pkg-config
 , nodePackages
-, src
-, version
 }:
 
 let
   prisma = nodePackages.prisma;
 in
 
-buildNpmPackage {
+buildNpmPackage rec {
   pname = "pingvin-share-backend";
-  inherit version;
+  version = "0.22.2";
 
-  src = "${src}/backend";
+  src = fetchFromGitHub {
+    owner = "stonith404";
+    repo = "pingvin-share";
+    rev = "v${version}";
+    hash = "sha256-OR+5AN3LYA4D+xY3KO8BloDqflBlB+ZNSIcXKEOYiL0=";
+  };
+
+  sourceRoot = "source/backend";
 
   npmInstallFlags = [ "--build-from-source" ];
   installPhase = ''
diff --git a/pkgs/servers/pingvin-share/default.nix b/pkgs/servers/pingvin-share/default.nix
index b178bdd977c0..f1395908a222 100644
--- a/pkgs/servers/pingvin-share/default.nix
+++ b/pkgs/servers/pingvin-share/default.nix
@@ -1,23 +1,12 @@
 { callPackage
-, fetchFromGitHub
 , recurseIntoAttrs
 , nixosTests
 }:
 
-let
-  version = "0.22.2";
-  src = fetchFromGitHub {
-    owner = "stonith404";
-    repo = "pingvin-share";
-    rev = "v${version}";
-    hash = "sha256-OR+5AN3LYA4D+xY3KO8BloDqflBlB+ZNSIcXKEOYiL0=";
-  };
-in
-
-recurseIntoAttrs {
-  backend = callPackage ./backend.nix { inherit src version; };
+recurseIntoAttrs rec {
+  backend = callPackage ./backend.nix { };
 
-  frontend = callPackage ./frontend.nix { inherit src version; };
+  frontend = callPackage ./frontend.nix { inherit (backend) src version; };
 
   passthru.tests = {
     pingvin-share = nixosTests.pingvin-share;
diff --git a/pkgs/servers/pingvin-share/frontend.nix b/pkgs/servers/pingvin-share/frontend.nix
index 0a4c944a37b9..c19e26ceced7 100644
--- a/pkgs/servers/pingvin-share/frontend.nix
+++ b/pkgs/servers/pingvin-share/frontend.nix
@@ -8,9 +8,9 @@
 
 buildNpmPackage {
   pname = "pingvin-share-frontend";
-  inherit version;
+  inherit src version;
 
-  src = "${src}/frontend";
+  sourceRoot = "source/frontend";
 
   npmInstallFlags = [ "--build-from-source" ];
   installPhase = ''
```

Reply:
Looks like everything else has been addressed.

Reply:
Though it seems like the update bot _can_ handle `version`/`src` in a `let` block. Example: https://github.com/NixOS/nixpkgs/commit/3b4473b04d9ef01bc4dc8f885efeb34bbfbed850

So maybe this is not an issue?

Reply:
Sorry I was busy IRL so I couldn't give much time to NixOS, but now it's Ok so I can update this package and try to finish for a merge

Reply:
I just noticed that the reason `nixpkgs-update` works for `memos` is because it has a custom `updateScript`. I don't think the bot is gonna work even if we move `src` and `version` into the backend package.

The bot will update the backend's `npmDepsHash`, but it won't update the frontend's `npmDepsHash` so it will leave the package in a broken state. The only way to fix that as far as I'm aware is with a custom `updateScript`.

Personally I think this is out of scope for this MR. However I don't have merge rights so I gotta leave this up to @SuperSandro2000 decide